internal/git/git.go

```diff
@@ -141,16 +141,25 @@ func PathWithNameSpace(remote string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	parts := strings.Split(remoteURL, "/")
-	l := len(parts)
-	if l < 2 {
+
+	parts := strings.Split(remoteURL, "//")
+
+	if len(parts) == 1 {
+		// scp-like short syntax (e.g. git@gitlab.com...)
+		part := parts[0]
+		parts = strings.Split(part, ":")
+	} else if len(parts) == 2 {
+		// every other protocol syntax (e.g. ssh://, http://, git://)
+		part := parts[1]
+		parts = strings.SplitN(part, "/", 2)
+	} else {
 		return "", errors.Errorf("cannot parse remote: %s url: %s", remote, remoteURL)
 	}
 
-	path := strings.Join(parts[l-2:l], "/")
-	if i := strings.LastIndex(path, ":"); i != -1 {
-		path = path[i+1:]
+	if len(parts) != 2 {
+		return "", errors.Errorf("cannot parse remote: %s url: %s", remote, remoteURL)
```

Comment:
Shouldn't this be caught by the above else?

Reply:
Oops... Of course it will be.

Reply:
Wait, no! The `parts` is being reassigned inside the `if..else` statement, so this `if` checks the new variable. Turns out this is quite confusing though, but I was really struggling to find proper names for the variables, so just ended up reassigning.